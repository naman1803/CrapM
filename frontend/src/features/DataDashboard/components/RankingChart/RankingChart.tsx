import React, { useState } from 'react';
import {
    Box,
    Collapse,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableHead,
    TableRow,
    Typography,
} from '@mui/material';
import { FacilityExceedanceData } from '../../types/ChartData';
import { ResponsiveContainer } from 'recharts';

export interface RankingChartProps {
    data: FacilityExceedanceData[];
    children: React.ReactNode;
}

export function sortRankExceedanceElements(
    exceedanceData: FacilityExceedanceData[],
): FacilityExceedanceData[] {
    const uniqueExceedances: Record<string, number> = {};

    // compile all exceedances by facility owner name
    exceedanceData.forEach(({ facilityOwner, noOfExceedances }) => {
        if (uniqueExceedances[facilityOwner]) {
            uniqueExceedances[facilityOwner] += noOfExceedances;
        } else {
            uniqueExceedances[facilityOwner] = noOfExceedances;
        }
    });

    // compile unique set
    const elements: FacilityExceedanceData[] = Object.entries(uniqueExceedances).map(
        ([facilityOwner, noOfExceedances]) => ({ facilityOwner, noOfExceedances })
    );

    // sort by exceedance numbers
    elements.sort((a, b) => b.noOfExceedances - a.noOfExceedances);

    return elements;
}

const RankingChart: React.FC<RankingChartProps> = ({ data, children }) => {
    const [openRow, setOpenRow] = useState<number | null>(null);

    // sort data into rankings
    const uniqueData = sortRankExceedanceElements(data);

    // handler for opening detail boxes
    const handleRowClick = (index: number) => {
        setOpenRow(openRow === index ? null : index);
    };

    return (
        <ResponsiveContainer width="100%" height="100%" minHeight={'400px'}>
            <Box>
                {children}
                <TableContainer component={Box} sx={{ maxHeight: 400, overflow: 'auto' }}>
                    <Table stickyHeader aria-label="collapsible table">
                        <TableHead>
                            <TableRow>
                                <TableCell>Rank</TableCell>
                                <TableCell>Facility Owner</TableCell>
                            </TableRow>
                        </TableHead>
                        <TableBody>
                            {uniqueData.map((element, index) => (
                                <React.Fragment key={element.facilityOwner}>
                                    <TableRow
                                        hover
                                        onClick={() => handleRowClick(index)}
                                    >
                                        <TableCell>{index + 1}</TableCell>
                                        <TableCell>{element.facilityOwner}</TableCell>
                                    </TableRow>
                                    <TableRow>
                                        <TableCell colSpan={2} sx={{ p: 0 }}>
                                            <Collapse in={openRow === index} timeout="auto" unmountOnExit>
                                                <Box padding={'0.5rem'}>
                                                    <Typography variant="h6">Rating Details</Typography>
                                                    <Typography variant="body2">
                                                        Number of Exceedances: {element.noOfExceedances}
                                                    </Typography>
                                                </Box>
                                            </Collapse>
                                        </TableCell>
                                    </TableRow>
                                </React.Fragment>
                            ))}
                        </TableBody>
                    </Table>
                </TableContainer>
            </Box>
        </ResponsiveContainer>
    );
};

export default RankingChart;
