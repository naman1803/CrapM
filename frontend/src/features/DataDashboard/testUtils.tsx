jest.mock('recharts', () => ({
  __esModule: true,
  ...jest.requireActual('recharts'),
  ResponsiveContainer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>ResponsiveContainer</p>
      {children}
    </div>
  )),
}));
