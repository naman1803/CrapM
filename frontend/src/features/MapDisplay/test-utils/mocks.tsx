import {
  LakeGeoPoint,
  MapElement,
  MapElementType,
  SewageGeoPoint,
} from '../types/MapElements';

export const mockLakeElements: MapElement[] = [
  {
    id: 1,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      name: 'Grrrr Lake',
      additional_notes: 'This is a lake',
    } as LakeGeoPoint,
  },
  {
    id: 2,
    dataType: 'lake',
    isSelectable: true,
    elementType: MapElementType.Marker,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      name: 'Obama Lake',
    } as LakeGeoPoint,
  },
];

export const mockSewageElements: MapElement[] = [
  {
    id: 1,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 1,
      facility_owner: 'Eric',
      radius: 100,
      colour: 'Blue',
    } as SewageGeoPoint,
  },
  {
    id: 2,
    dataType: 'sewage',
    isSelectable: true,
    elementType: MapElementType.Circle,
    dataPoint: {
      coordinates: [0, 0],
      id: 2,
      facility_owner: 'Myron',
      radius: 100,
      colour: 'Red',
    } as SewageGeoPoint,
  },
];
