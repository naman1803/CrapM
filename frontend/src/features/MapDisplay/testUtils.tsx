import { LeafletEventHandlerFnMap, LatLngExpression } from 'leaflet';

jest.mock('react-leaflet', () => ({
  __esModule: true,
  ...jest.requireActual('leaflet'),
  useMap: () => ({
    flyTo: jest.fn(),
    getZoom: jest.fn().mockReturnValue(13), // Add getZoom method returning default zoom level
    on: jest.fn(), // Mock on method for event handling
    off: jest.fn(), // Mock off method for removing event listeners
  }),
  MapContainer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>Map Container</p>
      {children}
    </div>
  )),
  Marker: jest
    .fn()
    .mockImplementation(
      ({
        position,
        children,
        key,
        eventHandlers,
      }: {
        eventHandlers: LeafletEventHandlerFnMap;
        position: LatLngExpression;
        key: any;
        children: React.ReactNode;
      }) => (
        <div key={key}>
          <p>{`Marker-${position.toString()}`}</p>
          {Object.entries(eventHandlers).map(([keyE, value]) => (
            <button key={keyE} onClick={value}>
              {`button-${keyE}-${position.toString()}`}
            </button>
          ))}
          {children}
        </div>
      ),
    ),
  Popup: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>Popup</p>
      {children}
    </div>
  )),
  TileLayer: jest.fn().mockImplementation(({ children }) => (
    <div>
      <p>TileLayer</p>
      {children}
    </div>
  )),
}));
