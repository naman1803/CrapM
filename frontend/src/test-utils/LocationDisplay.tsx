import { useLocation } from 'react-router-dom';
/**test component which displays the current url location.
 * use testid = location-display to get value of current location
 */
const LocationDisplay: React.FC = () => {
  const location = useLocation();

  return <div data-testid='location-display'>{location.pathname}</div>;
};
export default LocationDisplay;
