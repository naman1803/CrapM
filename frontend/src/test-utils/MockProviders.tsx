import { ThemeProvider } from '@mui/material';
import { MemoryRouter } from 'react-router-dom';
import { themeOptions } from '@app/theme';
import React from 'react';

// Wraps all providers for the app for testing
export const TestProviders = ({
  children,
  initialRoute = '/',
}: {
  children?: React.ReactNode;
  initialRoute?: string;
}) => {
  return (
    <ThemeProvider theme={themeOptions}>
      <MemoryRouter initialEntries={[initialRoute]}>{children}</MemoryRouter>
    </ThemeProvider>
  );
};
