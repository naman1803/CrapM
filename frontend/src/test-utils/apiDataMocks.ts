import { LakeData } from '@/types/LakeData';
import {
  SewageCoordData,
  SewageData,
  SewageSeverityData,
} from '@/types/SewageData';

export const mockSewageSeverityElement: SewageSeverityData = {
  id: 1,
  facilityOwner: 'Eric',
  contaminants: {
    brainrot: {
      averageSeverityScore: 1.0,
      overallSeverityLevel: 'Moderate',
      averagePercentExceedance: 100.0,
      records: [
        {
          severityScore: 1.0,
          severityLevel: 'Moderate',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
    takingDumb: {
      averageSeverityScore: 2.0,
      overallSeverityLevel: 'High',
      averagePercentExceedance: 200.0,
      records: [
        {
          severityScore: 0.5,
          severityLevel: 'High',
          date: '2021-01-01',
        },
        {
          severityScore: 2.0,
          severityLevel: 'High',
          date: '2021-01-02',
        },
      ],
    },
  },
};

export const mockLakeApiData: LakeData[] = [
  {
    id: 1,
    lakeName: 'Lake Ontario',
    latitude: 45.2,
    longitude: -78.94,
    additionalNotes: null,
  },
  {
    id: 2,
    lakeName: 'Lake Erie',
    latitude: 46.47,
    longitude: -82.78,
    additionalNotes: null,
  },
];

export const mockSewageApiDataSent = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    contaminant: 'PH - HIGH',
    radius: 22.524761,
    colour: "Green"
  },
];
export const mockSewageApiData: SewageCoordData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    latitude: 44.49,
    longitude: -77.69,
    radius: 22.539892,
    colour: "Green"
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    latitude: 44.68,
    longitude: -79.23,
    radius: 22.524761,
    colour: "Green"
  },
];

export const mockSewageSeverityApiData: SewageSeverityData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    contaminants: {
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.77,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 177.0,
        records: [
          {
            severityScore: 0.77,
            severityLevel: 'Low',
            date: '2016-08-18',
          },
        ],
      },
    },
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    contaminants: {
      'PH - HIGH': {
        averageSeverityScore: 0.95,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 195.4,
        records: [
          {
            severityScore: 0.95,
            severityLevel: 'Low',
            date: '2016-08-30',
          },
        ],
      },
      'BIOCHEMICAL OXYGEN DEMAND': {
        averageSeverityScore: 0.58,
        overallSeverityLevel: 'Low',
        averagePercentExceedance: 157.8,
        records: [
          {
            severityScore: 0.58,
            severityLevel: 'Low',
            date: '2016-08-23',
          },
        ],
      },
    },
  },
];

export const mockSewageAllApiData: SewageData[] = [
  {
    id: 1,
    facilityOwner: 'Mock Facility 1',
    siteAddress: '123 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-18T00:00:00',
    exceedanceEndDate: '2016-08-18T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 88.5,
    quantityMinimum: null,
  },
  {
    id: 2,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'BIOCHEMICAL OXYGEN DEMAND',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-23T00:00:00',
    exceedanceEndDate: '2016-08-23T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 78.9,
    quantityMinimum: null,
  },
  {
    id: 3,
    facilityOwner: 'Mock Facility 2',
    siteAddress: '1234 Mock St',
    contaminant: 'PH - HIGH',
    typeOfExceedance: 'Legislation Non-Compliance',
    exceedanceStartDate: '2016-08-30T00:00:00',
    exceedanceEndDate: '2016-08-30T00:00:00',
    contaminantLimit: 50.0,
    contaminantUnit: 'mg/L',
    limitFrequency: 'any',
    noOfExceedances: 1,
    quantityMaximum: 97.7,
    quantityMinimum: null,
  },
];
