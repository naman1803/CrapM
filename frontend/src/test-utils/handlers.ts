import { http, HttpResponse } from 'msw';
import {
  mockLakeApiData,
  mockSewageAllApiData,
  mockSewageApiDataSent,
  mockSewageSeverityApiData,
} from './apiDataMocks';

const handlers = [
  http.get('/api/maps/lakes/all', () => {
    return HttpResponse.json(mockLakeApiData);
  }),
  http.get('/api/maps/sewage/all', () => {
    return HttpResponse.json(mockSewageApiDataSent);
  }),
  http.get('/api/data/sewage/all', () => {
    return HttpResponse.json(mockSewageAllApiData);
  }),
  ...mockSewageSeverityApiData.map((severityData) =>
    http.get(`/api/data/sewage/severity/${severityData.id}`, () => {
      return HttpResponse.json(severityData);
    }),
  ),
];
export default handlers;
