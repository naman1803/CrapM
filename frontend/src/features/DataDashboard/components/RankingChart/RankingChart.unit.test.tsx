import './../../testUtils';

import { TestProviders } from "@/test-utils/MockProviders";
import { render } from "@testing-library/react";
import RankingChart, { RankingChartProps } from './RankingChart';
import Typography from '@mui/material/Typography';

const mockRankingChartData: RankingChartProps = {
    data: [
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 3 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 7 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 5 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 2 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 9 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 6 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 8 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 1 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 4 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 10 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 5 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 6 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 3 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 7 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 8 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 4 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 2 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 9 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 1 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 5 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 10 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 3 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 4 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 7 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 2 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 5 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 8 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 1 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 9 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 6 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 3 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 10 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 2 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 5 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 7 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 6 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 4 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 1 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 8 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 3 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 10 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 9 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 2 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 7 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 6 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 4 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 1 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 8 },
        { facilityOwner: 'Northland Waterworks Ltd', noOfExceedances: 5 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 3 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 10 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 9 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 6 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 2 },
        { facilityOwner: 'Eastern Transit Solutions', noOfExceedances: 4 },
        { facilityOwner: 'Southern Logistics Co.', noOfExceedances: 10 },
        { facilityOwner: 'Central Railways Inc.', noOfExceedances: 8 },
        { facilityOwner: 'Pacific Maritime LLC', noOfExceedances: 9 },
        { facilityOwner: 'Green Valley Utilities', noOfExceedances: 5 },
        { facilityOwner: 'Western Cargo Group', noOfExceedances: 6 },
    ],
    children: <Typography>Hello World</Typography>
};

describe('RankingChart', () => {
    it('should render the chart', () => {
        const { asFragment } = renderRankingChart(mockRankingChartData);
        expect(asFragment()).toMatchSnapshot();
    });
    // it('should display the first cell component as the highest \'exceedant\' child', () => {
    //     const { container } = renderRankingChart(mockRankingChartData);
    //     expect(getByText('Hello World')).toBeInTheDocument();
    //     const firstElement = container.querySelector('');
    //     expect(lineTag).toBeInTheDocument();
    // });
    it('should display its child components', () => {
        const { getByText } = renderRankingChart(mockRankingChartData);
        expect(getByText('Hello World')).toBeInTheDocument();
    });
});

const renderRankingChart = (props: RankingChartProps) => render(
    <TestProviders>
        <RankingChart {...props} />
    </TestProviders>
);
