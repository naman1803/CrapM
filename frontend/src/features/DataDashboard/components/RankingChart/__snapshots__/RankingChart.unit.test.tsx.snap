// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RankingChart should render the chart 1`] = `
<DocumentFragment>
  <div>
    <p>
      ResponsiveContainer
    </p>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-tckkrt-MuiTypography-root"
      >
        Hello World
      </p>
      <div
        class="MuiTableContainer-root MuiBox-root css-cwly5v-MuiTableContainer-root"
      >
        <table
          aria-label="collapsible table"
          class="MuiTable-root MuiTable-stickyHeader css-1q0iqtv-MuiTable-root"
        >
          <thead
            class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-2o0037-MuiTableRow-root"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-7owr83-MuiTableCell-root"
                scope="col"
              >
                Rank
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-7owr83-MuiTableCell-root"
                scope="col"
              >
                Facility Owner
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                1
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Central Railways Inc.
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                2
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Green Valley Utilities
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                3
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Northland Waterworks Ltd
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                4
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Pacific Maritime LLC
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                5
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Southern Logistics Co.
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                6
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Eastern Transit Solutions
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                7
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ltj5m4-MuiTableCell-root"
              >
                Western Cargo Group
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-2o0037-MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1tbxb7l-MuiTableCell-root"
                colspan="2"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
